lib/v2/buaa/sme.js

```diff
@@ -0,0 +1,184 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+
+const BASE_URL = 'http://www.sme.buaa.edu.cn';
+
+module.exports = async (ctx) => {
+    const { type = 'tzgg' } = ctx.params;
+    const { name, path } = getType(type);
+    const list = await getList(`${BASE_URL}/${path}.htm`);
+    ctx.state.data = {
+        // 源标题
+        title: `${name} - 集成电路学院`,
+        // 源链接
+        link: `http://www.sme.buaa.edu.cn/tzgg.htm`,
+        // 源文章
+        item: await getItems(ctx, list),
+    };
+};
+
+async function getList(url) {
+    const { data } = await got(url);
+    const $ = cheerio.load(data);
+    const list = $("div[class='Newslist'] > ul > li")
+        .map((_, item) => {
+            item = $(item);
+            const $a = item.find('a');
+            const link = $a.attr('href');
+            return {
+                title: item.find('a').text(),
+                link: link.startsWith('http') ? link : `${BASE_URL}/${link}`, // 有些链接是相对路径
+                pubDate: timezone(parseDate(item.find('span').text()), +8),
+            };
+        })
+        .get();
+    return list;
+}
+
+function getItems(ctx, list) {
+    return Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const { data: descrptionResponse } = await got(item.link);
+                const $descrption = cheerio.load(descrptionResponse);
+                item.description = $descrption('div[class="v_news_content"]').html();
+                return item;
+            })
+        )
+    );
+}
+
+function getType(type) {
+    const res = {
+        name: type,
+        path: type,
+    };
+    switch (type) {
+        // 首页
+        case 'xyyw':
+            res.name = '学院要闻';
+            break;
+        case 'tzgg':
+            res.name = '通知公告';
+            break;
+        case 'cggs':
+            res.name = '采购公示';
+            break;
+        case 'xsbg':
+            res.name = '学术报告';
+            break;
+        case 'jxxx':
+            res.name = '教学信息';
+            break;
+        // 学院概况
+        case 'xkjs':
+            res.name = '学科建设';
+            res.path = 'xygk/xkjs';
```

Comment:
You can let user input `xygk/xkjs` directly instead of `xkjs` which you don't need to update this mapping in the future.